src/compiler/types/lines_8000_end.rs

```diff
@@ -19,12 +19,62 @@ pub struct PrinterOptions {}
 pub trait EmitTextWriter: SymbolWriter {
     fn write(&mut self, s: &str);
     fn write_trailing_semicolon(&mut self, text: &str);
+    fn write_comment(&mut self, text: &str);
     fn get_text(&self) -> String;
+    fn raw_write(&mut self, text: &str);
+    fn write_literal(&mut self, text: &str);
+    fn get_text_pos(&self) -> usize;
+    fn get_line(&self) -> usize;
+    fn get_column(&self) -> usize;
+    fn get_indent(&self) -> usize;
+    fn is_at_start_of_line(&self) -> bool;
+    fn has_trailing_comment(&self) -> bool;
+    fn has_trailing_whitespace(&self) -> bool;
+    fn get_text_pos_with_write_line(&self) -> Option<usize> {
+        None
+    }
+    fn non_escaping_write(&mut self, text: &str) {}
 }
 
 pub trait ModuleSpecifierResolutionHost {}
 
-pub trait SymbolTracker {}
+pub trait SymbolTracker {
+    fn track_symbol(
+        &mut self,
+        symbol: &Symbol,
+        enclosing_declaration: Option<Rc<Node>>,
+        meaning: SymbolFlags,
+    ) -> Option<bool> {
+        None
+    }
+    fn report_inaccessible_this_error(&mut self) {}
```

Comment:
For optional interface methods that return `void` or an `Option`, seems like giving a default trait method implementation is a nice way of not having to write out the "dummy" method implementations in individual implementors